cumulus/steps/ec2/alb.py

```diff
@@ -0,0 +1,135 @@
+        asg_sg = CLUSTER_SG_NAME % self.name
+        alb_sg = ALB_SG_NAME % self.name
+
+        # TODO: this SG shouldn't be required.. test.
+        template.add_resource(ec2.SecurityGroup(
+            asg_sg,
+            GroupDescription=asg_sg,
+            VpcId=Ref("VpcId")))
+
+        # ALB Security group
+        template.add_resource(
+            ec2.SecurityGroup(
+                alb_sg,
+                GroupDescription=alb_sg,
+                VpcId=Ref("VpcId")
+            ))
+
+        template.add_output(
+            Output("InternalAlbSG", Value=Ref(alb_sg))
+        )
+
+        # TODO: take a list of Cidr's
+        # Allow Internet to connect to ALB
+        template.add_resource(ec2.SecurityGroupIngress(
+            "LocalNetworkTo%sAlbPort443" % self.name,
+            IpProtocol="tcp", FromPort="443", ToPort="443",
+            CidrIp="10.0.0.0/0",
+            GroupId=Ref(alb_sg),
+        ))
+
+    def create_load_balancer_alb(self, template):
+        alb_name = ALB_NAME % self.name
+        alb_sg = ALB_SG_NAME % self.name
+
+        load_balancer = template.add_resource(alb.LoadBalancer(
+            alb_name,
+            # Name=alb_name,
+            Scheme="internal",
+            Subnets=Ref("PrivateSubnets"),
+            SecurityGroups=[Ref(alb_sg)]
+        ))
+
+        template.add_output(
+            Output(
+                "CanonicalHostedZoneID",
+                Value=load_balancer.GetAtt("CanonicalHostedZoneID")
+            )
+        )
+        template.add_output(
+            Output("DNSName", Value=load_balancer.GetAtt("DNSName"))
+        )
+
+    def add_listener(self, template):
+        # Choose proper certificate source ?-> always acm?
+        acm_cert = Join("", [
+            "arn:aws:acm:",
+            Ref("AWS::Region"),
+            ":",
+            Ref("AWS::AccountId"),
+            ":certificate/", Ref("ALBCertName")])
+        # We probably don't need this code for an IAM Cert
+        iam_cert = Join("", [
+            "arn:aws:iam::",
+            Ref("AWS::AccountId"),
+            ":server-certificate/",
+            Ref("ALBCertName")])
+        cert_id = If("UseIAMCert", iam_cert, acm_cert)
+        alb_name = ALB_NAME % self.name
+
+        with_ssl = alb.Listener(
+            "Listener",
+            Port="443",
+            Protocol="HTTPS",
+            LoadBalancerArn=Ref(alb_name),
+            DefaultActions=[alb.Action(
+                Type="forward",
+                TargetGroupArn=Ref(TARGET_GROUP_DEFAULT % self.name)
+            )],
+            Certificates=[alb.Certificate(
+                CertificateArn=cert_id
+            )]
+        )
+
+        template.add_resource(with_ssl)
+
+        template.add_output(
+            Output("IAlbListener", Value=with_ssl.Ref())
+        )
+
+    def create_default_target_group(self, template):
+        template.add_resource(alb.TargetGroup(
+            TARGET_GROUP_DEFAULT % self.name,
```

Comment:
If we run two instances of this, ie a bswift and a sauter instance separately,  this looks like we'll get name collisions.  Prefix support in cumulus is probably a bigger discussion.